src/app/services/covid-data-api.service.ts

```diff
@@ -28,14 +28,14 @@ export class CovidDataApiService implements CovidDataApiServiceInterface {
   ) {}
 
   /**
-   * Cycles through each api service, calling the method provided, till one returns a value
+   * Calls all api services and returns the first non-error response
```

Comment:
Not sure if that's the best approach, we'll end up making a lot of pointless api calls.

We could try something like this - suppose for the sake of illustration that we have an api which has a `call(successCallback, errorCallback)` method, where callbacks are the functions to be called on success failure (you can tailor it to the actual api later)

Then if we have an array of apis we can do
```
callApis = apis.reduceRight( 
  (result, api) => () => api.call(successCallback, result), 
  () => console.error('All api calls failed')
);
callApis();
```
The way this works is that you're reducing on the error function (void function with no arguments) and you recursively put it inside the error callback.

Reply:
I'm not sure I'm following what you meant in the snippet of code you have here. At least unsure of what you mean with using reduce/reduceRight.

Could be being a bit slow today so will try take a look at it another day.

Or could fix this part up in another PR (the way we have it now passes the tests, but then again those may not cover everything). Just to get everything else in so that if you or duncan want to work on other stuff that is dependent on this, then you can.

Reply:
Perhaps it will be helpful to trace the reduce cumulant on an example. Say you have an array `[api1, api2]`.
The cumulant is:

Start:
```
  () => console.error('All api calls failed')
```

Step 1:
```
() => api2.call(
     successCallback, 
     () => console.error('All api calls failed')
  )
```

Step 2:
```
() => api1.call(
  successCallback,
  () => api2.call(
       successCallback, 
       () => console.error('All api calls failed')
    )
```

So you have this cascade of api calls where if the first succeeds it just calls successCallback, but if it fails it then calls api2 which does the same etc. and in the end you just get an error message.

I'm happy to include it as a separate PR.